Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -402,73 +428,87 @@ - (void)startObservingLookupDismissalIfNeeded
 
 - (void)_windowDidOrderOnScreen:(NSNotification *)notification
 {
-    _impl->windowDidOrderOnScreen();
+    if (_impl)
```

Comment:
My understanding is if `_windowDidOrderOnScreen` is called, then `WKWindowVisibilityObserver` is observing `_window` and `_window` is non-null, then `_impl` cannot be null. Is this correct? If so, it seems we should assert instead of early return?

Reply:
Well, in theory, yes. _impl shouldn't be null here but given we want to put this fix into a SU very late, I'm making it as conservative as possible.

Reply:
Technically if `_impl` can be null, then `_window` is likely null, and we will be not removing observer (as we currently use `_window` to decide whether we should remove observer in `stopObserving`), which might still lead to crash...